src/test/store/profile-view.test.js

```diff
@@ -1933,6 +1933,69 @@ describe('snapshots of selectors/profile', function () {
   });
 });
 
+describe('changeSelectedCallNode', function () {
+  it('state is kept separately for inverted and non-inverted views', function () {
```

Comment:
Thanks for the new test!
I'm not sure this tests that state is kept separately, but it tests that the behavior exposed by this works fine. Maybe the title could be adjusted? (optional)

Reply:
Changed the title to: "switching between the call tree and the flame graph always selects a reasonable node"